GIT_STYLE_GUIDE.md

```diff
@@ -0,0 +1,273 @@
+  </thead>
+  <tbody>
+    <tr>
+      <td>Master</td>
+      <td>master</td>
+      <td>Accepts merges from Develop and Hotfixes</td>
+    </tr>
+    <tr>
+      <td>Develop</td>
+      <td>develop</td>
+      <td>Accepts merges from us/Issues and Hotfixes</td>
+    </tr>
+    <tr>
+      <td>Task</td>
+      <td>task/short-desc</td>
+      <td>Always branch off HEAD of develop, it referes to an iteration task described on Trello.</td>
+    </tr>
+    <tr>
+      <td>Hotfix</td>
+      <td>hotfix/short-desc</td>
+      <td>Always branch off Master, receives utmost priority for development, testing and code reviews. After merging and fixing Master in production it's changes get merged into Develop</td>
+    </tr>
+    <tr>
+      <td>Issue</td>
+      <td>issue/{{issue-id}}-short-desc</td>
+      <td>Never branches off Master, it may branch from any other branch as necessary but primarily branches off  develop.</td>
+    </tr>
+    <tr>
+      <td>Docs</td>
+      <td>docs/short-desc</td>
+      <td>Branches off develop, and will contain documentation that does not refer to any specific piece of code i.e. this guide.</td>
+    </tr>
+  </tbody>
+</table>
+
+* Choose *short* and *descriptive* names:
+
+  ```shell
+  # good
+  $ git checkout -b task/oauth-migration
+
+  # bad - too vague
+  $ git checkout -b task/login_fix
+  ```
+
+* Always follow the folder structure outlined above:
+
+  ```shell
+  # good
+  $ git checkout -b issue/18-utf-enconding
+
+  #bad
+  $ git checkout -b fix-utf-encoding
+  ```
+
+* Always branch off from the allowed branches for each instance, as specified above:
+
+  ``` shell
+  #good
+  $ git checkout -b task/database-table-creation develop
+
+  #bad
+  $ git checkout -b task/database-table-creation master
+  ```
+
+* Use *dashes* to separate words.
+
+* Delete your branch from the upstream repository after it's merged, unless
+  there is a specific reason not to.
+
+ [How to delete a branch via de terminal](https://makandracards.com/makandra/621-git-delete-a-branch-local-or-remote)
+ [How to delete a branch from the upstream repository via github's GUI](https://github.com/blog/1377-create-and-delete-branches)
+
+  Tip: Use the following command while being on "master", to list merged
+  branches:
+
+  ```shell
+  $ git branch --merged | grep -v "\*"
+  ```
+
+# Commits
+
+* Each commit should be a single *logical change*. Don't make several
+  *logical changes* in one commit. For example, if a patch fixes a bug and
+  optimizes the performance of a feature, split it into two separate commits.
+
+* Don't split a single *logical change* into several commits.
+
+* Commit *early* and *often*. Small, self-contained commits are easier to
+  understand and revert when something goes wrong.
+
+* Before pushing make sure, your code passes your tests and is properly linted.
+
+## Messages
+
+* Write your commit messages in imperative present tense.
+
+* Use the editor, not the terminal, when writing a commit message:
```

Comment:
Just as @csssaz says, it is actually there on the examples. If you are not familiar with vi, or you would like to use another editor with git you can change it this way.

https://help.github.com/articles/associating-text-editors-with-git/

It could be added to the style guide if you like, or think that many people will have trouble with this as well.